keras_preprocessing/image/image_data_generator.py

```diff
@@ -322,10 +322,11 @@ def __init__(self,
         self.std = None
         self.principal_components = None
 
-        if np.isscalar(zoom_range):
+        if isinstance(zoom_range, float):
             self.zoom_range = [1 - zoom_range, 1 + zoom_range]
         elif len(zoom_range) == 2:
-            self.zoom_range = [zoom_range[0], zoom_range[1]]
+            if all(isinstance(val, float) for val in zoom_range):
+                self.zoom_range = [zoom_range[0], zoom_range[1]]
```

Comment:
This can silently fail if zoom_range is not made of floats. We should raise an error

Reply:
Isn't that what happens with the last `else` statement? It is my understanding that an error will be raised if `zoom_range` is not made of floats.

Reply:
Should be `elif len(zoom_range) == 2 and all(isinstance(val, float) for val in zoom_range)`
then. Otherwise it won't get inside the `else`.

Reply:
That slipped by me. I'll incorporate your feedback into the fix. Thank you.